sample_analyze_read.py

```diff
@@ -0,0 +1,192 @@
+    #     poller = document_intelligence_client.begin_analyze_document(
+    #         "prebuilt-read",
+    #         analyze_request=f,
+    #         features=[DocumentAnalysisFeature.STYLE_FONT],
+    #         content_type="application/octet-stream",
+    #     )
+    result: AnalyzeResult = poller.result()
+    
+    # [START analyze_read]
+    # Detect languages.
+    print("----Languages detected in the document----")
+    if result.languages is not None:
+        for language in result.languages:
+            print(f"Language code: '{language.locale}' with confidence {language.confidence}")
+
+    # Detect styles.
+    print("----Styles detected in the document----")
+    if result.styles:
+        for style in result.styles:
+            if style.is_handwritten:
+                print("Found the following handwritten content: ")
+                print(",".join([result.content[span.offset : span.offset + span.length] for span in style.spans]))
+            if style.font_style:
+                print(f"The document contains '{style.font_style}' font style, applied to the following text: ")
+                print(",".join([result.content[span.offset : span.offset + span.length] for span in style.spans]))
+    
+    # To learn the detailed concept of "bounding polygon" in the following content, visit: https://aka.ms/bounding-region
+    # Analyze pages.
+    for page in result.pages:
+        print(f"----Analyzing document from page #{page.page_number}----")
+        print(f"Page has width: {page.width} and height: {page.height}, measured with unit: {page.unit}")
+
+        # Analyze lines.
+        if page.lines:
+            for line_idx, line in enumerate(page.lines):
+                words = get_words(page, line)
+                print(
+                    f"...Line # {line_idx} has {len(words)} words and text '{line.content}' within bounding polygon '{line.polygon}'"
+                )
+
+                # Analyze words.
+                for word in words:
+                    print(f"......Word '{word.content}' has a confidence of {word.confidence}")
+
+        # Analyze selection marks.
+        if page.selection_marks:
+            for selection_mark in page.selection_marks:
+                print(
+                    f"...Selection mark is '{selection_mark.state}' within bounding polygon "
+                    f"'{selection_mark.polygon}' and has a confidence of {selection_mark.confidence}"
+                )
+        # Note that selection marks returned from begin_analyze_document(model_id="prebuilt-read") do not return the text associated with the checkbox. 
+        # For the API to return this information, build a custom model to analyze the checkbox and its text. For detailed steps, visit: https://aka.ms/train-your-custom-model
+        
+    # Analyze paragraphs.
+    if result.paragraphs:
+        print(f"----Detected #{len(result.paragraphs)} paragraphs in the document----")
+        for paragraph in result.paragraphs:
```

Comment:
can you please double check if paragraph.role applies to read model output? My understanding is this only applies to layout model.

Reply:
https://learn.microsoft.com/en-us/azure/ai-services/document-intelligence/concept-read?view=doc-intel-4.0.0#paragraphs
The Read concept page shows the extraction of the paragraph, and the output of the code is the same as the "output" display in the document.